pkg/security/podsecuritypolicy/provider.go

```diff
@@ -160,7 +160,7 @@ func (s *simpleProvider) CreateContainerSecurityContext(pod *api.Pod, container
 	// if we're using the non-root strategy set the marker that this container should not be
 	// run as root which will signal to the kubelet to do a final check either on the runAsUser
 	// or, if runAsUser is not set, the image UID will be checked.
-	if sc.RunAsNonRoot == nil && s.psp.Spec.RunAsUser.Rule == extensions.RunAsUserStrategyMustRunAsNonRoot {
+	if sc.RunAsNonRoot == nil && sc.RunAsUser == nil && s.psp.Spec.RunAsUser.Rule == extensions.RunAsUserStrategyMustRunAsNonRoot {
```

Comment:
what is the purpose for this?  I think we lose the intention/protection of the setting via the kubelet if we're conditionally setting it.  I guess the argument is that if something later sets `runAsUser` then it could also set `runAsNonRoot` to false so it really isn't protection anyway?

Reply:
Today, a `MustRunAsNonRoot` PSP would mutate a pod where all the containers had `runAsNonRoot:null, runAsUser:5`, which is unnecessary (they all are explicitly not running as root already), breaks update cases (since validation would prevent mutation of the spec), and breaks detection of which PSPs would allow the pod as-is

Reply:
If the container already sets an explicit runAsUser that is non-zero, there's no kubelet-side protection we need